common/models/container.js

```diff
@@ -294,44 +307,56 @@ module.exports = function(Container) {
               }
             })
 
-
+            //this gets processed before the previous then block is completed
+            //TODO: promisify it properly
```

Comment:
Have you already created a JIRA issue to track this if you want to fix it on a later date?

Reply:
It comes under the general "clean code" task. This TODO will just come handy when we are cleaning code in general